### DISCUSSION TITLE
I can't tell if this is a bug or not?

### DISCUSSION THREAD

**Discussion by lloydjatkinson** [snarfed/bridgy#1795] [General]
Hi! I am totally new to indieweb so sorry for any confusion.

I am trying to implement webmention support on my site. I have setup https://webmention.io/ and have various HTML and microformat classes on my site. The docs on this are pretty vague and it's not really clear why duplication is needed (such as the last two lines).

```tsx
<link rel="webmention" href="https://webmention.io/www.lloydatkinson.net/webmention" />
<a rel="me" href="https://mastodon.social/@lloydjatkinson" class="hidden">Mastodon</a>
<a rel="me" href="https://github.com/lloydjatkinson" class="hidden">GitHub</a>
<a rel="author" class="p-author h-card hidden" href="https://www.lloydatkinson.net/">Lloyd Atkinson</a>
<a href="https://www.lloydatkinson.net/" class="h-card hidden" rel="me">Lloyd Atkinson</a>
```

For the first several hours none of these would work, it said "no webmention support".
![image](https://github.com/user-attachments/assets/310681b5-0725-4e4a-83a4-424549fd7212)

Now, they appear to be working without any code changes on my site.
![image](https://github.com/user-attachments/assets/258e3a58-4e8b-46d1-a85f-735ecccf47f4)

But the real issue I'm facing is this error when I try "resend for post":

![image](https://github.com/user-attachments/assets/d94e4b67-9137-4559-8830-daba8ce4ac0c)

I want to see if there are any webmentions for a specific URL (eg https://www.lloydatkinson.net/posts/2024/frictions-and-complexities-of-simple-bash-scripts/) and when I paste that in the text input and click "Discover" I get that error. Which Mastodon link is it not finding? How do I add this link?

This is made even more confusing because when I then try "Discover" with a URL brid.gy already knows about, I get the same error? If it can't find the link how was it able to find any responses to begin with?

**Comment by lloydjatkinson** [comment]
Also, as you can see here I can see "in-reply-to" but no "like-of"?

https://webmention.io/api/mentions.jf2?target=https://www.lloydatkinson.net/posts/2024/stupid-problems-require-stupid-solutions-cloudflare-is-breaking-my-svgs/

Yet brid.gy sees likes: 
![image](https://github.com/user-attachments/assets/2226540c-e5d7-4751-acd5-5663d5495ddb)



And doesn't seem to be sending them to webmentions.io? These are all replies, not likes.

![image](https://github.com/user-attachments/assets/5203f59f-7318-4abb-abb9-7bc82ea16e44)

**Comment by snarfed** [comment]
Hey, sorry for the confusion!

> I want to see if there are any webmentions for a specific URL (eg https://www.lloydatkinson.net/posts/2024/frictions-and-complexities-of-simple-bash-scripts/) and when I paste that in the text input and click "Discover" I get that error. Which Mastodon link is it not finding? How do I add this link?

Bridgy is looking for a syndication link, ie with the `u-syndication` mf2 class. More: https://brid.gy/about#post+have+a+link , https://brid.gy/about#link , https://indieweb.org/rel-syndication#How_to_publish

If you enter a Mastodon post URL into Bridgy's "Resend for post" form, and that post links to your site, Bridgy will find that connection. To find it starting from your site, it looks for a syndication link.

> And doesn't seem to be sending them to webmentions.io? These are all replies, not likes.

Hmm, looking at the brid.gy source URLs for those webmentions, I see a like and a repost. Maybe try a tool like https://pin13.net/mf2/ to inspect their mf2?

**Reply by lloydjatkinson** [reply]
Thanks for the tips! I will try to inspect the mf2 with that tool.

About syndication links, is there a good workflow for this? Because from my understanding it seems like a chicken and egg situation. I don't know ahead of time what the syndication link is going to be (because it's usually some random number, right?), so do I publish a post, then post it to Mastodon etc, and then go back and edit the post to contain a link with `u-syndication` class to that new Mastodon post?

I hope that makes sense 😄

**Reply by snarfed** [reply]
Right! _If_ you want both a syndication link in your web post and a link to your web post in your Mastodon post, then yes, that's a circular dependency. "Post web, post Mastodon, update web," like you describe, is the standard way to handle that.

Bridgy only needs one or the other to do backfeed, though, not both. So, up to you.

**Reply by lloydjatkinson** [reply]
Thanks for confirming my thinking 😄  You say it only needs one or the other, but should I do both anyway so that the "Discover" functionality will work without error?

**Reply by snarfed** [reply]
Discover can work either direction. If your Mastodon post has a link, you can give it your Mastodon post. If your web post has a synd link, give it your web post.
